app/crud/versions.py

```diff
@@ -100,6 +99,18 @@ async def create_version(dataset: str, version: str, **data) -> ORMVersion:
         )
         new_version.metadata = metadata
 
+    # FIXME: Should we really allow one to specify a new version as
+    # latest on creation? I thought we didn't. Seems like it will cause
+    # requests to temporarily go to a perhaps incompletely-imported
+    # asset...
+    if data.get("is_latest"):
+        await _reset_is_latest(dataset, version)
+        # logger.info(
+        #     f"Setting version {version} to latest for dataset {dataset}. "
+        #     f"Cache info: {get_latest_version.cache_info()}"
+        # )
+        _: bool = get_latest_version.cache_invalidate(dataset)
```

Comment:
Seems like it might make sense to  put the `cache_invalidate` call in `_reset_is_latest`, so we don't forget to call it when we call `reset_is_latest`?

Reply:
Yeah, I'll do that. I was on the fence about it, but now that you say it, that seems better.